titus-server-federation/src/main/java/com/netflix/titus/federation/service/FallbackJobServiceGateway.java

```diff
@@ -1,64 +1,254 @@
+        Mono<Void> primaryMono = primary.updateJobDisruptionBudget(request, callMetadata);
+        Mono<Void> secondaryMono = secondary.updateJobDisruptionBudget(request, callMetadata);
+        return getFallbackMono(methodName, primaryMono, secondaryMono);
+    }
+
+    @Override
+    public Mono<Void> updateJobAttributes(JobAttributesUpdate request, CallMetadata callMetadata) {
+        final String methodName = "updateJobAttributes";
+        Mono<Void> primaryMono = primary.updateJobAttributes(request, callMetadata);
+        Mono<Void> secondaryMono = secondary.updateJobAttributes(request, callMetadata);
+        return getFallbackMono(methodName, primaryMono, secondaryMono);
+    }
+
+    @Override
+    public Mono<Void> deleteJobAttributes(JobAttributesDeleteRequest request, CallMetadata callMetadata) {
+        final String methodName = "deleteJobAttributes";
+        Mono<Void> primaryMono = primary.deleteJobAttributes(request, callMetadata);
+        Mono<Void> secondaryMono = secondary.deleteJobAttributes(request, callMetadata);
+        return getFallbackMono(methodName, primaryMono, secondaryMono);
+    }
+
+    @Override
+    public Observable<Job> findJob(String jobId, CallMetadata callMetadata) {
+        final String methodName = "findJob";
+        Observable<Job> primaryObservable = primary.findJob(jobId, callMetadata);
+        Observable<Job> secondaryObservable = secondary.findJob(jobId, callMetadata);
+        return getFallbackObservable(methodName, primaryObservable, secondaryObservable);
+    }
+
+    @Override
+    public Observable<JobQueryResult> findJobs(JobQuery jobQuery, CallMetadata callMetadata) {
+        final String methodName = "findJobs";
+        Observable<JobQueryResult> primaryObservable = primary.findJobs(jobQuery, callMetadata);
+        Observable<JobQueryResult> secondaryObservable = secondary.findJobs(jobQuery, callMetadata);
+        return getFallbackObservable(methodName, primaryObservable, secondaryObservable);
+    }
+
+    @Override
+    public Observable<JobChangeNotification> observeJob(String jobId, CallMetadata callMetadata) {
+        final String methodName = "observeJob";
+        Observable<JobChangeNotification> primaryObservable = primary.observeJob(jobId, callMetadata);
+        Observable<JobChangeNotification> secondaryObservable = secondary.observeJob(jobId, callMetadata);
+        return getFallbackObservable(methodName, primaryObservable, secondaryObservable);
+    }
+
+    @Override
+    public Observable<JobChangeNotification> observeJobs(ObserveJobsQuery query, CallMetadata callMetadata) {
+        final String methodName = "observeJobs";
+        Observable<JobChangeNotification> primaryObservable = primary.observeJobs(query, callMetadata);
+        Observable<JobChangeNotification> secondaryObservable = secondary.observeJobs(query, callMetadata);
+        return getFallbackObservable(methodName, primaryObservable, secondaryObservable);
+    }
```

Comment:
This strategy will not work well with stream APIs. If there is an error after the stream is established and some items were already sent, we want to propagate the error, not redo the stream on the secondaryObservables